src/tomato/drivers/biologic/main.py

```diff
@@ -106,18 +183,18 @@ def get_data(
     with FileLock(lockpath, timeout=60) as fh:
         try:
             logger.info(f"connecting to '{address}:{channel}'")
-            id_, device_info = api.Connect(address)
+            id_, device_info = safe_api_connect(api, address)
             logger.info(f"getting data from '{address}:{channel}'")
             data = api.GetData(id_, channel)
             logger.info(f"disconnecting from '{address}:{channel}'")
-            api.Disconnect(id_)
+            safe_api_disconnect(api, id_)
         except Exception as e:
             logger.critical(f"{e=}")
+            raise
```

Comment:
Why?

Reply:
My bad, I initially thought of also raising runtimeerror here saying fail to connect/disconnect. Then I realized such clause should be in the safe api connect/disconnect. So, I rewmoved the exception here but forgot to remove raise. I will clean this up.